src/analysis/retail/priest/discipline/modules/spells/TwilightEquilibrium.tsx

```diff
@@ -85,22 +111,28 @@ class TwilightEquilibrium extends Analyzer {
     }
   }
 
+  onPTWApply(event: CastEvent | RefreshDebuffEvent | ApplyDebuffEvent) {
+    const targetString = encodeEventTargetString(event) || '';
+    if (this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_HOLY_BUFF.id)) {
+      this.ptwTargets.add(targetString);
+    } else {
+      this.ptwTargets.delete(targetString);
+    }
```

Comment:
You implied earlier that the TE boost stays for the duration of the DoT, even through refreshes. If so, you need special case handling here that will not delete from the Set if the event is a refresh.

That would be odd behavior though - the boost really can't be downgraded?